### PR TITLE
Bug 1169373 — Log roomConnectionId on /rooms/:roomToken status update calls.

### PR DESCRIPTION
**Pull request by Natim**
https://bugzilla.mozilla.org/show_bug.cgi?id=1169373

### CODE REVIEW

File: loop/routes/rooms.js [hunk 1]
```diff
@@ -472,7 +472,17 @@ module.exports = function (apiRouter, conf, logError, storage, filestorage, auth
         handleUpdateStatus: function(req, res) {
           // Room status update is validated and logged in middlewares
           validators.validateRoomStatusUpdate(req, res, function () {
-            res.status(204).json();
+            storage.getRoomParticipant(req.token, participantHmac,
+              function(err, participant) {
+                if (res.serverError(err)) return;
```
Comment (leplatrem) [inline loop/routes/rooms.js]: Deserves a test ?

File: loop/routes/rooms.js [hunk 2]
```diff
@@ -472,7 +472,17 @@ module.exports = function (apiRouter, conf, logError, storage, filestorage, auth
...
+                if (participant === null) {
+                  sendError(res, 400, errors.NOT_ROOM_PARTICIPANT,
+                            "Unable to leave a room you have not joined.");
+                  return;
```
Comment (leplatrem) [inline loop/routes/rooms.js]: Deserves a test ?

Comment (Natim) [inline loop/routes/rooms.js]: Also the message doesn't looks right here.

File: loop/routes/rooms.js [hunk 3]
```diff
@@ -472,7 +472,17 @@ module.exports = function (apiRouter, conf, logError, storage, filestorage, auth
         handleUpdateStatus: function(req, res) {
           // Room status update is validated and logged in middlewares
           validators.validateRoomStatusUpdate(req, res, function () {
-            res.status(204).json();
+            storage.getRoomParticipant(req.token, participantHmac,
+              function(err, participant) {
+                if (res.serverError(err)) return;
+                if (participant === null) {
+                  sendError(res, 400, errors.NOT_ROOM_PARTICIPANT,
+                            "Unable to leave a room you have not joined.");
```
Comment (adamroach) [inline loop/routes/rooms.js]: The error message here should be adjusted ("Can't update status for a room you aren't in.")

File: test/rooms_test.js [hunk 4]
```diff
@@ -1702,6 +1702,15 @@ describe("/rooms", function() {
...
         });
 
+        it("should log the roomConnectionId", function(done) {
+          updateStateRoom(hawkCredentials, roomToken).end(function(err) {
+            if (err) throw err;
+            expect(logs).to.length(3);
+            expect(logs[2].roomConnectionId).to.not.be.undefined;
+            done();
+          });
+        });
```
Comment (adamroach) [inline test/rooms_test.js]: I think I'd expect to see the rainy-day case (i.e., user not in room) tested here as well.

**Comment by Natim** [comment]
Updated.

**Comment by leplatrem** [comment]
r+

**Pull request merged**